deep_qa/layers/wrappers/multi_input_time_distributed.py

```diff
@@ -0,0 +1,161 @@
+from keras import backend as K
+from keras.layers import InputSpec, TimeDistributed
+from deep_qa.layers.wrappers.fixed_time_distributed import FixedTimeDistributed
+from overrides import overrides
+
+
+class MultiInputTimeDistributed(FixedTimeDistributed):
+    r"""
+    This class extends the previous TimeDistributed fix (i.e., ``FixedTimeDistributed``)
+    to allow for inputs that are lists of tensors.  Currently, while masking is supported masking (i.e., if any of the inputs
+    has a mask) is not supported.
+
+    Inputs:
+        - probabilities, shape ``(batch, ..., N, ...)``,
+          with an optional mask of the same shape,
+          where N is the number of y's in the above equation
+          (i.e. the number of probabilities being combined in the product),
+          in the dimension corresponding to the specified axis.
+    Output:
+        - X, shape ``(batch, ..., ...)``, one less than input dims,
+          with an optional mask of the same shape.  The lost dimension corresponds to the specified axis.
+          The output mask is the result of `K.any()` on the input mask, along the specified axis.
+    Parameters
+    ----------
+    axis : int, default=-1
+        The axis over which to combine probabilities.
+    name : string, default='noisy_or'
+        Name of the layer, ued to debug both the layer and its parameter.
+    param_init : string, default='uniform'
+        The initialization of the noise parameter.
+    noise_param_constraint : Keras Constraint, default=None
+        Optional, a constraint which would be applied to the noise parameter.
+    """
+
+    @overrides
+    def build(self, input_shape):
+        num_inputs = len(input_shape)
```

Comment:
You should check for tuple vs. list here, like you do in `get_output_shape_for`.